llvm/lib/CodeGen/MachineVerifier.cpp

```diff
@@ -1966,6 +1969,11 @@ void MachineVerifier::visitMachineInstrBefore(const MachineInstr *MI) {
     if (SrcReg.isVirtual() && DstReg.isPhysical() && SrcSize.isScalable() &&
         !DstSize.isScalable())
       break;
+    // If the Src is scalable and the Dst is fixed, then Dest can only hold
```

Comment:
I accidentally "squashed" my cherry-picked commits from #70882 and [this commit](https://github.com/llvm/llvm-project/commit/4832c88e4956b3c7dc0a5064d946477448d5239d) together. 
I was trying to use rebase interactive to extract them apart but couldn't figure out a neat way. Any suggestion?

Reply:
One easy option when things get messy is to create a new branch from main, cherry pick the commits this is stacked on that has not been committed, and then cherry pick the changes this patch introduces on its own. Then you can replace riscv-v-add-gisel with the new branch with `git branch -m riscv-v-add-gisel riscv-v-add-gisel-bak && git branch -m your-new-branch riscv-v-add-gisel`. You will have to force push to this branch, but that should be okay since this PR is still a draft. It would be nice if we end up with a commit history that looks like this before moving out of draft:

```
head-commit-on-main
stacked-commit-1
...
stacked-commit-n
single-commit-that-this-pr-introduces
```